src/cutadapt/script.sh

```diff
@@ -190,10 +190,24 @@ debug
 # individual files.
 ###########################################################
 
-if [[ -z $par_fasta ]]; then
-  ext="fastq"
+# Extract extension from output pattern, preserving compression if specified
+echo ">> Determining output file extension from pattern: $par_output"
+
+# Extract everything after the last '.' in the filename part of the pattern
+# This handles patterns like "*.fastq.gz", "*.fasta.bz2", etc.
+filename=$(basename "$par_output")
```

Comment:
The `else` block defined defaults based on the fasta flag. But when does this `else` block ever get executed.?

On an unrelated note:  I am also a bit confused about the usage of the `output` argument. From my understanding it acts as a filter for the output files (it does not _change_ the file names)?  But this requires the user to know what the format is of the output files? And this depends on the other settings.